vividus-engine/src/main/java/org/vividus/log/LoggingTableTransformerMonitor.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.log;
+
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jbehave.core.model.ExamplesTable.TableProperties;
+import org.jbehave.core.model.TableParsers;
+import org.jbehave.core.model.TableTransformerMonitor;
+import org.jbehave.core.model.TableTransformers.Formatting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggingTableTransformerMonitor implements TableTransformerMonitor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingTableTransformerMonitor.class);
+
+    private static final Formatting FORMATTING = new Formatting();
+    private static final int MAX_ROWS_LOGGING = 50;
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    private final TableParsers tableParsers;
+
+    public LoggingTableTransformerMonitor(TableParsers tableParsers)
+    {
+        this.tableParsers = tableParsers;
+    }
+
+    @Override
+    public void beforeTransformerApplying(String transformerName, TableProperties properties, String inputTable)
+    {
+        logTransformerInfo(transformerName, properties, inputTable, "Input table body");
+    }
+
+    @Override
+    public void afterTransformerApplying(String transformerName, TableProperties properties, String outputTable)
+    {
+        logTransformerInfo(transformerName, properties, outputTable, "Output (transformed) table");
+    }
+
+    private void logTransformerInfo(String transformerName, TableProperties tableProperties, String table,
+            String tableType)
+    {
+        StringBuilder formatBuilder = new StringBuilder("Applying table transformer: {}").append(LINE_SEPARATOR)
+                .append("Properties: {}").append(LINE_SEPARATOR)
+                .append(String.format("%s:%n{}", tableType));
+        String format = formatBuilder.toString();
+        String propertiesForLogging = tableProperties.getPropertiesAsString().trim();
+        String tableForLogging = getExampleTableForLogging(table, tableProperties);
+
+        LOGGER.info(format, transformerName, propertiesForLogging, tableForLogging);
+    }
+
+    private String getExampleTableForLogging(String exampleTableOrigin, TableProperties tableProperties)
+    {
+        if (StringUtils.isBlank(exampleTableOrigin))
+        {
+            return "EMPTY";
```

Comment:
```suggestion
            return ">>>  EMPTY  <<<";
```
to make it more unique